tincr/io/device/cell_library.tcl

```diff
@@ -470,38 +481,34 @@ proc create_leaf_cell_pin_mapping_permutable {cell bel xml_out} {
     place_cell $cell $bel
     
     set pin_map [dict create]
-    
-    set input_bel_pin_names [list] 
-    foreach input_bel_pin [get_bel_pins -of $bel -filter {DIRECTION==IN} -quiet] {
-        lappend input_bel_pin_names [lindex [split [get_property NAME $input_bel_pin] "/"] end]
-    }
-    
+    set input_bel_pins [get_bel_pins -of $bel -filter {DIRECTION==IN} -quiet]
+   
     foreach cell_pin [get_pins -of $cell] {
         
-        set is_input [expr {[get_property DIRECTION $cell_pin] == "IN"}]
-       
-        set cell_pin_name [lindex [split $cell_pin "/"] end]
         set bel_pins [get_bel_pins -of $cell_pin -quiet]
         
-        if {$is_input} {
-            ::tincr::assert { [llength $bel_pins] == 0 } "\[Assertion Error\] Input pin to permutable cell should have no default pin mapping $cell/$cell_pin_name [get_sites -of $cell]"
-            foreach bel_pin_name $input_bel_pin_names {    
-                
+        if {[get_property DIRECTION $cell_pin] == "IN"} {
+            ::tincr::assert { [llength $bel_pins] == 0 } "\[Assertion Error\] Input pin to permutable cell should have no default pin mapping $cell/$cell_pin [get_sites -of $cell]"
+            
+            set cell_pin_name [lindex [split $cell_pin "/"] end]
+            foreach bel_pin $input_bel_pins {    
+                set bel_pin_name [lindex [split [get_property NAME $bel_pin] "/"] end]
                 if { [catch {[set_property LOCK_PINS "{$cell_pin_name:$bel_pin_name}" $cell]} err] == 0 } {
-                    dict lappend pin_map $cell_pin_name $bel_pin_name
+                    dict lappend pin_map $cell_pin $bel_pin
                 }
                 reset_property LOCK_PINS $cell
             }
         } else { ; # output cell pin
-            ::tincr::assert { [llength $bel_pins] == 1 } "\[Assertion Error\] An output pin should map to exactly one bel pin. $cell/$cell_pin_name [get_sites -of $cell]"
-            set bel_pin_name [lindex [split $bel_pins "/"] end]
-            dict lappend pin_map $cell_pin_name $bel_pin_name
+            ::tincr::assert { [llength $bel_pins] == 1 } "\[Assertion Error\] An output pin of a permutable cell should map to exactly one bel pin. $cell_pin [get_sites -of $cell]"
+            
+            dict lappend pin_map $cell_pin [lindex $bel_pins 0]
```

Comment:
Does `tincr::assert` throw when the statement evaluates to false? If not, it should. If so, this line will never be reached.